drivers/serial/uart_gecko.c

```diff
@@ -18,6 +18,10 @@
 #include <em_gpio.h>
 #endif /* CONFIG_PINCTRL */
 
+#ifdef CONFIG_PM_DEVICE
+#include <zephyr/pm/device.h>
+#endif
```

Comment:
Normally, the header should written to be included unconditionally (if #ifdef are needed, they should be in the header file). 

So, unless they are not present, I would prefer to include headers unconditionally.